src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -8,153 +8,106 @@
 using System.Security.Cryptography.X509Certificates;
 using Microsoft.Win32.SafeHandles;
 
+using SafeSslHandle = Interop.libssl.SafeSslHandle;
+
 internal static partial class Interop
 {
     internal static class OpenSsl
     {
-        #region structures
-        [StructLayout(LayoutKind.Sequential)]
-        private struct SslContext
-        {         
-            internal IntPtr sslPtr;
-            internal IntPtr readBioPtr;
-            internal IntPtr writeBioPtr;
-            internal bool isServer;
-        }
-        #endregion
-
         #region internal methods
 
-
         //TODO (Issue #3362) Set remote certificate options
-        internal static IntPtr AllocateSslContext(long options, SafeX509Handle certHandle, SafeEvpPKeyHandle certKeyHandle, bool isServer, bool remoteCertRequired)        
+        internal static SafeSslHandle AllocateSslContext(long options, SafeX509Handle certHandle, SafeEvpPKeyHandle certKeyHandle, bool isServer, bool remoteCertRequired)
         {
-            SslContext sslContext = new SslContext
-            {
-                isServer = isServer,
-            };
-
-            try
-            {
-                IntPtr method = GetSslMethod(isServer, options);
+            SafeSslHandle context = null;
 
-                IntPtr contextPtr = libssl.SSL_CTX_new(method);
+            IntPtr method = GetSslMethod(isServer, options);
 
-                if (IntPtr.Zero == contextPtr)
+            using (libssl.SafeSslContextHandle innerContext = new libssl.SafeSslContextHandle(method))
```

Comment:
This will Dispose of innerContext at the end of the using block.  That's ok?  I see it's being passed to methods like SafeSslHandle.Create... it's not needed after those calls?  Or they're AddRef'ing to keep it alive?